core/primitives/src/runtime/parameter_table.rs

```diff
@@ -254,72 +308,20 @@ impl ParameterTable {
         cost: near_primitives_core::config::ActionCosts,
     ) -> Result<near_primitives_core::runtime::fees::Fee, InvalidConfigError> {
         let key: Parameter = format!("{}", FeeParameter::from(cost)).parse().unwrap();
-        let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-        value.as_fee().ok_or(InvalidConfigError::WrongValueType(
-            key,
-            std::any::type_name::<Fee>(),
-            value.clone(),
-        ))
-    }
-
-    /// Read and parse a number parameter from the `ParameterTable`.
-    fn get_number<T>(&self, key: Parameter) -> Result<T, InvalidConfigError>
-    where
-        T: TryFrom<u64>,
-        T::Error: Into<std::num::TryFromIntError>,
-    {
-        let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-        let value_u64 = value.as_u64().ok_or(InvalidConfigError::WrongValueType(
-            key,
-            std::any::type_name::<T>(),
-            value.clone(),
-        ))?;
-        T::try_from(value_u64).map_err(|err| {
-            InvalidConfigError::WrongIntegerType(
-                err.into(),
-                key,
-                std::any::type_name::<T>(),
-                value_u64,
-            )
-        })
-    }
-
-    /// Read and parse a u128 parameter from the `ParameterTable`.
-    fn get_u128(&self, key: Parameter) -> Result<u128, InvalidConfigError> {
-        let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-        value.as_u128().ok_or(InvalidConfigError::WrongValueType(
-            key,
-            std::any::type_name::<u128>(),
-            value.clone(),
-        ))
+        self.get(key)
     }
 
     /// Read and parse a string parameter from the `ParameterTable`.
     fn get_account_id(&self, key: Parameter) -> Result<AccountId, InvalidConfigError> {
-        let value = self.parameters.get(&key).ok_or(InvalidConfigError::MissingParameter(key))?;
-        let value_string = value.as_str().ok_or(InvalidConfigError::WrongValueType(
-            key,
-            std::any::type_name::<String>(),
-            value.clone(),
-        ))?;
-        value_string.parse().map_err(|err| {
+        let value: &str = self.get(key)?;
+        value.parse().map_err(|err| {
```

Comment:
This could also be a `TryFrom` implementation.

Reply:
True, I initially thought this would lead to a worse error message, but in the end this was not the case. Done.